### PR TITLE
feat: deploy guac with postgres backend

### PR DESCRIPTION
**Pull request by dejanb**
This is the first deployment of Guac with PSQL.

To use:

1. Copy database to the `deploy/compose/container_files/pg` with appropriate privileges set
2. I used `--db-migrate=false` to make it work here. You can try without it for probably better performance (here queries take a few seconds on the first call)
3. The good way to test queries are by running these tests https://github.com/trustification/trustification/blob/main/spog/api/src/service/guac.rs#L220. That is what would guac-rs return now to the spog-api

### CODE REVIEW

File: deploy/compose/.env [hunk 1]
```diff
@@ -13,7 +13,7 @@ COLLECTORIST_API_PORT=8180
 COLLECTOR_OSV_API_PORT=8181
 COLLECTOR_SNYK_API_PORT=8182
 COLLECTOR_NVD_API_PORT=8183
-GUAC_IMAGE=ghcr.io/trustification/guac:v0.3.13-temp
+GUAC_IMAGE=quay.io/mrizzi/guac:20231107-111157-arm64
```
Comment (mrizzi) [inline deploy/compose/.env:16]: `amd64` images
```suggestion
GUAC_IMAGE=quay.io/mrizzi/guac:20231107-134115-amd64
```

**Comment by dejanb** [comment]
One thing we can test at the moment, is "related products" query

Get the token:
```
TOKEN=$(curl -s -d "client_id=walker" -d "client_secret=ZVzq9AMOVUdMY1lSohpx1jI3aW56QDPS" -d 'grant_type=client_credentials' \
  'http://localhost:8090/realms/chicken/protocol/openid-connect/token' | jq -r .access_token)
```

Call the API
```
$curl -v -X GET --oauth2-bearer $TOKEN "http://localhost:8083/api/v1/cve/cve-2022-2284/related-products" | jq
  ```

**Comment by dejanb** [comment]
Here are complete instructions for how to make `find_vulnerability` query running


1. Download, unzip and copy Guac database from [here](https://drive.google.com/file/d/1enZDAq0gxsH9EZZr3Mh1ES-4bQY6Q6nj/view?usp=sharing) to `deploy/compose/container_files/pg`
2. In `deploy/compose/compose_guac.yaml`, I used --db-migrate=false to make it work here. You can try without it for probably better performance (here queries take a few seconds on the first call)
3. Start core services `docker compose -f compose.yaml -f compose-guac.yaml up --force-recreate`
4.  Start trustification services
```
RUST_LOG=info cargo run -p trust -- vexination api --devmode &
RUST_LOG=info cargo run -p trust -- bombastic api --devmode &
RUST_LOG=info cargo run -p trust -- spog api --devmode &
RUST_LOG=info cargo run -p trust -- v11y api --devmode &
RUST_LOG=info cargo run -p trust -- vexination indexer --devmode &
RUST_LOG=info cargo run -p trust -- bombastic indexer --devmode &
RUST_LOG=info cargo run -p trust -- v11y indexer --devmode &
```
5. Start walkers
```
RUST_LOG=info cargo run -p trust bombastic walker --sink http://localhost:8082 --devmode -3
RUST_LOG=info cargo run -p trust -- v11y walker --devmode --source ../cvelistV5/
```
6. After the walkers have been working for a while, you can run the query like
```
$ TOKEN=$(curl -s -d "client_id=walker" -d "client_secret=ZVzq9AMOVUdMY1lSohpx1jI3aW56QDPS" -d 'grant_type=client_credentials' \
  'http://localhost:8090/realms/chicken/protocol/openid-connect/token' | jq -r .access_token)

$ curl -v -X GET --oauth2-bearer $TOKEN "http://localhost:8083/api/v1/sbom/vulnerabilities?id=cert-manager-1.json.bz2" | jq
```

Next steps:

1. Figure out the paging
2. Investigate what we need for `backtraces` information from guac
3. Investigate if we can run with collectors running and call `VulnEqual` before calling v11y

**Comment by dejanb** [comment]
Implemented pagination for the query and API calls. Now we can call it like

```
curl -v -X GET --oauth2-bearer $TOKEN "http://localhost:8083/api/v1/sbom/vulnerabilities?id=rhel-7.9.z.json.bz2&offset=0&limit=100" | jq
```

Defaults for the page options are `None`, so it will probably hang for big SBOMs. Maybe some sane defaults are in order?

File: deploy/compose/.env (hunk 1 above)
Comment (dejanb) [reply deploy/compose/.env]: We now have an "official" image that solves this.

File: deploy/k8s/chart/templates/guac/graphql/030-Deployment.yaml [hunk 2]
```diff
@@ -37,14 +37,67 @@ spec:
...
           env:
             - name: NAMESPACE
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
             {{ include "trustification.http-server" . | nindent 12 }}
+            - name: PGHOST
+              valueFrom:
+                secretKeyRef:
+                  name: guac-admin-db
```
Comment (ctron) [inline deploy/k8s/chart/templates/guac/graphql/030-Deployment.yaml:61]: Please to not use the `admin` connection. That should be the `user` connection.

File: deploy/openshift/template.yaml [hunk 3]
```diff
@@ -1287,6 +1297,46 @@ objects:
                   value: /etc/tls/tls.key
                 - name: HTTP_SERVER_TLS_CERTIFICATE_FILE
                   value: /etc/tls/tls.crt
+                - name: PGHOST
+                  valueFrom:
+                    secretKeyRef:
+                      name: guac-admin-db
```
Comment (ctron) [inline deploy/openshift/template.yaml:1303]: Same here.

**Review by ctron — CHANGES_REQUESTED** [review]

File: deploy/k8s/chart/templates/guac/graphql/030-Deployment.yaml (hunk 2 above)
Comment (dejanb) [reply deploy/k8s/chart/templates/guac/graphql/030-Deployment.yaml]: Fixed

File: deploy/openshift/template.yaml (hunk 3 above)
Comment (dejanb) [reply deploy/openshift/template.yaml]: Fixed

**Review by lulf — APPROVED** [review]

**Pull request merged**